src/pages/Home.tsx

```diff
@@ -1,80 +1,159 @@
 import "./Home.css";
-import { useState } from 'react';
+import { useEffect, useState } from "react";
 import axios from "axios";
+import { Button, TextField, Dialog } from "@mui/material";
 
-export default function Home(){
-    const [isModalOpen, setIsModalOpen] = useState(false);
-    const [formData, setFormData] = useState({
-        username: '',
-        email: '',
-        password: ''
+export default function Home() {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [formData, setFormData] = useState({
+    username: "",
+    email: "",
+    password: "",
+  });
+  const [userNameError, setUserNameError] = useState("");
+  const [isPasswordValid, setIsPasswordValid] = useState("");
+
+  const openModal = () => setIsModalOpen(true);
+  const closeModal = () => {
+    setIsModalOpen(false);
+    setUserNameError("");
+    setIsPasswordValid("");
+    setFormData({
+      username: "",
+      email: "",
+      password: "",
     });
+  };
+
+  // callbacks
+  const onChange = (e: any) => {
+    const { name, value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  const isAlphaNumeric = (data: any) => {
+    if (data === "") return true;
+    const pattern =
+      /^(?!.*[_.]{2})[a-zA-Z0-9](?!.*[_.]$)[a-zA-Z0-9_.]{1,18}[a-zA-Z0-9]$/;
+    return pattern.test(data);
+  };
 
-    const openModal = () => setIsModalOpen(true);
-    const closeModal = () => setIsModalOpen(false);
+  const validateUserName = (username: any) => {
+    if (username.length < 3 || username.length > 50) {
+      setUserNameError(
+        "Username should be of minimum 3 letters and not more than 50."
+      );
+      return false;
+    } else if (!isAlphaNumeric(username)) {
```

Comment:
username Alphanumeric q rakhna h? username may inlcude letters and numbers but not necessary